trin-core/src/portalnet/overlay_service.rs

```diff
@@ -640,7 +645,7 @@ where
                         None,
                         Some(query_id),
                     );
-                    let _ = self.request_tx.send(request);
+                    let _ = self.command_tx.send(OverlayCommand::Request(request));
```

Comment:
Can you help me understand why this change is needed or better? Something feels wrong about them both being valid options. Or is this also fixing some kind of bug?

Reply:
Upon investigation, that was the third commit making up for a mistake during a rebase on the first commit. I re-did the commits to fix that, thanks!